providers/structured/provider.go

```diff
@@ -5,118 +5,133 @@ package structured
 
 import (
 	"context"
+	"sync"
 
 	"github.com/myhelix/contextlogger/log"
 	"github.com/myhelix/contextlogger/providers"
 	"github.com/myhelix/contextlogger/providers/chaining"
 )
 
-type RawLogCallArgs struct {
+type LogCallArgs struct {
 	ContextFields log.Fields
 	Report        bool
 	Args          []interface{}
+	Level         providers.LogLevel
 }
 
 type RecordCallArgs struct {
 	ContextFields log.Fields
 	Metrics       log.Metrics
-}
-
-type RecordEventCallArgs struct {
-	ContextFields log.Fields
 	EventName     string
-	Metrics       log.Metrics
 }
 
 type StructuredOutputLogProvider struct {
 	providers.LogProvider
 
-	rawLogCalls      map[providers.RawLogCallType][]RawLogCallArgs
-	recordCalls      []RecordCallArgs
-	recordEventCalls []RecordEventCallArgs
-}
-
-func (p StructuredOutputLogProvider) GetRawLogCalls() map[providers.RawLogCallType][]RawLogCallArgs {
-	return p.rawLogCalls
+	logCalls    []*LogCallArgs
+	logMutex    sync.RWMutex
+	recordCalls []*RecordCallArgs
+	recordMutex sync.RWMutex
+}
+
+// Return list of log calls, filtered to only selected levels (if any present)
+func (p StructuredOutputLogProvider) LogCalls(levels ...providers.LogLevel) (result []*LogCallArgs) {
+	p.logMutex.RLock()
+	defer p.logMutex.RUnlock()
+
+	for _, call := range p.logCalls {
+		if len(levels) == 0 {
```

Comment:
@chriswhelix Can I put this `len(levels)` outside the for loop? Or is this a constant time operation?

Reply:
Should be very cheap. I'm not sure that making this a tiny bit faster is worth extra lines of code.